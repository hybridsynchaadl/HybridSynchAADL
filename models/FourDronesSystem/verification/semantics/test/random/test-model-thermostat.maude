set show timing off .

load ../../interpreter-random.maude
load ../syntax/module-thermostat-noparam.maude

---------------------------------------------
mod TEST-THERMOSTAT-SYSTEM is
  including THERMOSTAT-SYSTEM-NOPARAM .
  including MODEL-TRANSITION-SYSTEM .

  op initState : -> Object .
  eq initState = initialize(model) .
endm

rew {{initState,0} | 0 | 500} .

rew {{initState,400} | 0 | 500} .

rew {{initState,800} | 0 | 500} .

rew {{initState,1200} | 0 | 500} .
