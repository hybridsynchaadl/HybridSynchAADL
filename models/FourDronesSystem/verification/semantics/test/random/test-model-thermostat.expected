==========================================
rewrite in TEST-THERMOSTAT-SYSTEM : {{initState,0} | 0 | 500} .
rewrites: 74201
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [2.8786773271497058e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [2.7831977492719982e+1] # [true] >),connections
    : empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    2.9285867805598745e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,600} | 500 | 500}
==========================================
rewrite in TEST-THERMOSTAT-SYSTEM : {{initState,400} | 0 | 500} .
rewrites: 74155
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [2.7210460357254021e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [2.6595549170340405e+1] # [true] >),connections
    : empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    2.8188022686049358e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,1000} | 500 | 500}
==========================================
rewrite in TEST-THERMOSTAT-SYSTEM : {{initState,800} | 0 | 500} .
rewrites: 74201
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [3.0165015381179987e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [3.0109001399613774e+1] # [true] >),connections
    : empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    3.0038285048072762e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l2,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,1400} | 500 | 500}
==========================================
rewrite in TEST-THERMOSTAT-SYSTEM : {{initState,1200} | 0 | 500} .
rewrites: 74238
result GlobalSystem: {{< top : System | properties : TimingProperties::Period
    => {{5}},features : none,subcomponents : (< controller : Thread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    DataOutPort | properties : none,content : * # [false] > < off : DataOutPort
    | properties : none,content : * # [false] > < temp : DataInPort |
    properties : none,content : [0.0] # [false],cache : [2.9522628436880805e+1]
    # [true] >),subcomponents : (< lower : Data | properties :
    DataModel::InitialValue => {{10}},features : none,subcomponents : none,
    connections : empty,value : [1.0e+1] # [true] > < upper : Data | properties
    : DataModel::InitialValue => {{30}},features : none,subcomponents : none,
    connections : empty,value : [3.0e+1] # [true] > < curr : Data | properties
    : DataModel::InitialValue => {{0}},features : none,subcomponents : none,
    connections : empty,value : [2.8558971924781495e+1] # [true] >),connections
    : empty,varGen : < "top.controller" >,currState : l1,completeStates : l1,
    variables : x : Real,transitions : ((l1 -[on dispatch]-> l2 {(v{x} := ((c[
    curr] + f[temp]) / [[2]])) ; if(v[x] < c[lower])(on !) (elsif(v[x] > c[
    upper])off !) end if ; c{curr} := v[x]}) ; l2 -[otherwise]-> l1 {skip}),
    loopBound : 0,transBound : 2 > < environment : EnvWithTimer | properties :
    ((TimingProperties::Period => {{5}}) ; HybridSynchAADL::MaxClockDeviation
    => {{1.0000000000000001e-1}}),features : (< on : EnvInPort | properties :
    none,content : * # [false],envCache : null(Unit),target : controller > <
    off : EnvInPort | properties : none,content : * # [false],envCache : null(
    Unit),target : controller > < out : EnvOutPort | properties : none,content
    : [0.0] # [false],envCache : null(Real),target : controller >),
    subcomponents : < x : Data | properties : DataModel::InitialValue => {{
    16}},features : none,subcomponents : none,connections : empty,value : [
    2.9851043232104526e+1] # [true] >,connections : x ==> out,varGen : <
    "top.environment" >,currMode : l1,jumps : ((l1 -[on]-> l1) ; (l1 -[off]->
    l2) ; (l2 -[on]-> l1) ; l2 -[off]-> l2),flows : ((l1 [x(t)= (c[x] + (v[t] *
    [[2.0000000000000001e-1]]))]) ; l2 [x(t)= (c[x] - (v[t] * [[
    1.0000000000000001e-1]]))]),sampling : controller :(0,2),response :
    controller :(3,4),timer : 0,samplingIDs : controller,responseIDs :
    controller >),connections : (((controller .. on) --> (environment .. on)) ;
    ((controller .. off) --> (environment .. off)) ; (environment .. out) --> (
    controller .. temp)) >,1800} | 500 | 500}
Bye.
